File: family_order/src/web/com/asiainfo/veris/crm/order/web/group/param/adc/xfk/MebParamInfo.java

package com.asiainfo.veris.crm.order.web.group.param.adc.xfk;

import com.ailk.biz.view.IBizCommon;
import com.ailk.common.data.IData;
import com.ailk.common.data.IDataset;
import com.ailk.common.data.impl.DataMap;
import com.ailk.common.data.impl.DatasetList;
import com.asiainfo.veris.crm.order.pub.frame.bcf.util.IDataUtil;
import com.asiainfo.veris.crm.order.web.frame.bcf.base.CSViewCall;
import com.asiainfo.veris.crm.order.web.frame.csview.common.component.group.dynamichtml.IProductParamDynamic;
import com.asiainfo.veris.crm.order.web.frame.csview.common.svcutil.datainfo.userinfo.relationxxtinfo.RelationXXTInfoIntfViewUtil;

public class MebParamInfo extends IProductParamDynamic
{

    public IData getFamSnXxtDisByajax(IBizCommon bp, IData data) throws Throwable
    {
    	IData result = new DataMap();
    	IData results = new DataMap();
    	String serialNumber = data.getString("FAM_SN", "");
    	//true 订购过校讯通10套餐，false 没有订购过
    	boolean isXxtdisFlag = RelationXXTInfoIntfViewUtil.qryXFKMemInfoBySNandUserIdA(bp, serialNumber);
       result.put("isXxtDisFlag", isXxtdisFlag);
       results.put("AJAX_DATA", result);
       return results;


    }

    /**
     * 调用学护卡平台校验接口，校验号码是否已经录入
     *
     * @param bp
     * @param data
     * @return
     * @throws Throwable
     */
    public IData QryXfkNumIsExists(IBizCommon bp, IData data) throws Throwable
    {
        IData result = new DataMap();
        IData results = new DataMap();
        String serialNumber = data.getString("FAM_SN", "");
        IDataset selectedElement = new DatasetList(data.getString("SELECTED_ELEMENT"));
        String groupId = data.getString("GROUP_ID", "");
        boolean flag = true;
        String bizServCode = "";

        IData elementMap = null;
        IDataset attrParamList = null;
        IData attrParamMap = null;
        for(int i=0; i < selectedElement.size(); i++) {
            elementMap = selectedElement.getData(i);
            if ("574401".equals(elementMap.getString("ELEMENT_ID", ""))) {
                attrParamList = elementMap.getDataset("ATTR_PARAM");
                for (int j = 0; j < attrParamList.size(); j++) {
                    attrParamMap = attrParamList.getData(j);
                    if (!IDataUtil.isEmpty(attrParamMap.getData("PLATSVC"))) {
                        bizServCode = attrParamMap.getData("PLATSVC").getString("pam_BIZ_IN_CODE");
                    }
                }
            }
        }

        IData param = new DataMap();
        param.put("EC_ID", groupId);
        param.put("MOB_NUM", serialNumber);
        param.put("BIZ_SERV_CODE", bizServCode);
        param.put("OPR_CODE", "01");

        IDataset resultDs = CSViewCall.call(bp, "SS.XfkFamNumSynSVC.qryXfkNumIsExists", param);
        if(IDataUtil.isNotEmpty(resultDs) && ("0000".equals(resultDs.getData(0).getString("X_RESULTCODE")) || "00".equals(resultDs.getData(0).getString("X_RESULTCODE")))){
            flag = true;
        }else{
            flag = false;
        }
        result.put("isExistsFlag", flag);
        results.put("AJAX_DATA", result);
        return results;
    }

}

File: family_order/html/scripts/group/param/adc/xfk/MebParamInfo.js
/**
 * 作用：用来初始化页面的显示
 */
function init()
{

}

//点击下一步的校验
function validateParamPage()
{
    var famSn = $("#pam_NOTIN_FAM_SN").val();
    var grpUserId = $("#GRP_USER_ID").val();
    var groupId = $("#GROUP_ID").val();
    var mainNum = $("#MEB_SERIAL_NUMBER").val();

	var submitData = selectedElements.getSubmitData();
	var falg = false;
	var checkExists = false;

	for (var i = 0; i < submitData.length; i++)
	{
		if(submitData.get(i, "MODIFY_TAG")=="0" && submitData.get(i, "ELEMENT_ID")=="32000001"&& submitData.get(i, "ELEMENT_TYPE_CODE")=="D" )
	 	 {
	 		falg = true;
	 		break;
	 	 }

	}
	if(falg==true)
	{
	    if(""==famSn||null==famSn)
	    {
	    	alert("请添写家长号码！");
			return false;
	    }
		var isXxt = validateFamSn(famSn,grpUserId);//true 订购过和校园10套餐，false 没有订购过
		if(false==isXxt||"false"==isXxt)
		{
			alert("要订购学护卡15元包,家长号：【"+ famSn +"】必须先订购和校园10元套餐或者8元套餐 ！");
			return false;
		}
	}

	// 100036
	// 订购学护卡服务时先发起正向学护卡成员业务办理校验接口
	for (var i = 0; i < submitData.length; i++)
	{
		if(submitData.get(i, "MODIFY_TAG")=="0" && submitData.get(i, "ELEMENT_ID")=="574401"&& submitData.get(i, "ELEMENT_TYPE_CODE")=="S" )
	 	 {
			checkExists = true;
	 		break;
	 	 }

		if (submitData.get(i, "ELEMENT_ID")=="574401" && null!=famSn && ""!=famSn) {
			alert("校星卡服务不能填写家长号码！");
			return false;
		}

	}
//	if (checkExists==true) {
//		if(""==famSn||null==famSn)
//	    {
//	    	alert("家长号码不能为空，请填写！");
//			return false;
//	    }
//
//		var isExistsFamsn = validateFamSnIsExists(mainNum, submitData, groupId);
//
//		if(false==isExistsFamsn||"false"==isExistsFamsn)
//		{
//			alert("该号码尚未在学护卡平台录入资料，不能订购该业务 ！");
//			return false;
//		}
//	}

	return true;
}

function validateFamSn(famSn,grpUserId)
{
        var isXxt = false;
		var param = '&FAM_SN='+famSn+'&METHOD_NAME=getFamSnXxtDisByajax&CLASS_NAME=com.asiainfo.veris.crm.order.web.group.param.adc.xfk.MebParamInfo';
		Wade.httphandler.submit('','com.asiainfo.veris.crm.order.web.frame.csview.common.component.group.dynamichtml.ProxyParam','productParamInvoker',param,
	   	function(data){
	   		if(data != null){
				isXxt = getFamSnXxtDisByajax(data);//true 订购过和校园10套餐，false 没有订购过
			}
	    },
	    function(error_code,error_info,derror){
	   		showDetailErrorInfo(error_code,error_info,derror);
	    },
	    {async:false});

	    return isXxt;

}

//校验资费不允许在产品信息页面取消
function getFamSnXxtDisByajax(data)
{
	var isXxt = data.get("isXxtDisFlag");
    return isXxt
}

function validateFamSnIsExists(famSn,submitData, groupId)
{
        var isExists = false;
		var param = '&GROUP_ID='+groupId+'&FAM_SN='+famSn+'&SELECTED_ELEMENT='+submitData+'&METHOD_NAME=QryXfkNumIsExists&CLASS_NAME=com.asiainfo.veris.crm.order.web.group.param.adc.xfk.MebParamInfo';
		Wade.httphandler.submit('','com.asiainfo.veris.crm.order.web.frame.csview.common.component.group.dynamichtml.ProxyParam','productParamInvoker',param,
	   	function(data){
	   		if(data != null){
	   			isExists = getFamSnIsExistsByajax(data);//true 订购过和校园10套餐，false 没有订购过
			}
	    },
	    function(error_code,error_info,derror){
	   		showDetailErrorInfo(error_code,error_info,derror);
	    },
	    {async:false});

	    return isExists;

}

//校验资费不允许在产品信息页面取消
function getFamSnIsExistsByajax(data)
{
	var isExists = data.get("isExistsFlag");
    return isExists
}

